src/plugins/kibana/server/lib/create_mappings_from_pattern_fields.js

```diff
@@ -0,0 +1,38 @@
+const _ = require('lodash');
+
+// Creates an ES field mapping from a single field object in a kibana index pattern
+module.exports = function createMappingsFromPatternFields(fields) {
+  if (_.isEmpty(fields)) {
+    throw new Error('argument must not be empty');
+  }
+
+  const mappings = {};
+
+  _.forEach(fields, function (field) {
+    let mapping;
+
+    if (field.type === 'string') {
+      mapping = {
+        type: 'string',
+        index: 'analyzed',
+        omit_norms: true,
+        fielddata: {format: 'disabled'},
+        fields: {
+          raw: {type: 'string', index: 'not_analyzed', doc_values: true, ignore_above: 256}
+        }
+      };
+    }
+    else {
+      const fieldType = field.type === 'number' ? 'double' : field.type;
+      mapping = {
+        type: fieldType,
+        index: 'not_analyzed',
+        doc_values: true
+      };
+    }
+
+    _.set(mappings, field.name.replace('.', '.properties.'), mapping);
```

Comment:
replace only replaces the first instance of the pattern. I recommend using splice+join (which turns out to be more efficient as well as concise.

```js
field.name.split('.').join('.properties.')
```